File: bots/javascript/Bot1/Bot1/Bot.js
console.log("Bot2 ha iniziato l'esecuzione...");

function delay(time) {
    return new Promise(resolve => setTimeout(resolve, time));
}

async function executeBot() {
    // Fase 1
    console.log("Fase 1: Attesa per 30 secondi...");
    await delay(30000);
    console.log("Fase 1 completata!");

    // Fase 2
    console.log("Fase 2: Attesa per 30 secondi...");
    await delay(30000);
    console.log("Fase 2 completata!");

    // Fase 3
    console.log("Fase 3: Attesa per 1 minuto...");
    await delay(60000);
    console.log("Fase 3 completata!");

    console.log("Bot2 ha completato l'esecuzione.");
}

executeBot();

File: bots/java/Bot1/Bot.java
package Bot1;

public class Bot {
    public static void main(String[] args) {
        try {
            System.out.println("Bot1 ha iniziato l'esecuzione...");

            // Simula un'operazione lunga 30 secondi
            System.out.println("Fase 1: Attesa per 30 secondi...");
            Thread.sleep(30000);
            System.out.println("Fase 1 completata!");

            // Simula un'altra operazione lunga 30 secondi
            System.out.println("Fase 2: Attesa per 30 secondi...");
            Thread.sleep(30000);
            System.out.println("Fase 2 completata!");

            // Simula una fase finale
            System.out.println("Fase 3: Attesa per 1 minuto...");
            Thread.sleep(60000);
            System.out.println("Fase 3 completata!");

            System.out.println("Bot1 ha completato l'esecuzione.");
        } catch (InterruptedException e) {
            System.err.println("Errore durante l'esecuzione del bot.");
        }
    }
}